libs/libcommon/src/libcommon/queue.py

```diff
@@ -198,6 +202,53 @@ def flat_info(self) -> FlatJobInfo:
         )
 
 
+class Lock(Document):
+    meta = {"collection": QUEUE_COLLECTION_LOCKS, "db_alias": QUEUE_MONGOENGINE_ALIAS}
+    key = StringField(primary_key=True)
+    job_id = StringField()
+
+    created_at = DateTimeField(required=True)
+    updated_at = DateTimeField()
+
+    objects = QuerySetManager["Lock"]()
+
+
+class lock:
```

Comment:
the Lock objects are created and mutated, but never deleted. I think it's not an issue because the data is small, right?

Reply:
Yea we can figure deletion later - there won't be a ton of locks anyway.

The trivial implementation that deletes the lock on `.release()` doesn't work because another worker could try to acquire it at the same time.